packages/nvidia_nat_zep_cloud/src/nat/plugins/zep_cloud/zep_editor.py

```diff
@@ -16,90 +16,200 @@
         """
-
-        session_id = kwargs.pop("user_id")  # Ensure user ID is in keyword arguments
-        limit = top_k
-
-        search_result = await self._client.memory.search_sessions(session_ids=[session_id],
-                                                                  text=query,
-                                                                  limit=limit,
-                                                                  search_scope="messages",
-                                                                  **kwargs)
-
-        # Construct MemoryItem instances
-        memories = []
-
-        for res in search_result.results:
-            memories.append(
-                MemoryItem(conversation=[],
-                           user_id=session_id,
-                           memory=res.message.content,
-                           metadata={
-                               "relevance_score": res.score,
-                               "created_at": res.message.created_at,
-                               "updated_at": res.message.updated_at
-                           }))
-
-        return memories
+        user_id = kwargs.pop("user_id")  # Ensure user ID is in keyword arguments
+        mode = kwargs.pop("mode", "basic")  # Get mode, default to "basic" for fast retrieval
+
+        # Get thread_id from NAT context
+        thread_id = Context.get().conversation_id
+
+        # Fallback to default thread ID if no conversation_id available
+        if not thread_id:
+            thread_id = "default_zep_thread"
+
+        try:
+            # Use Zep v3 thread.get_user_context - returns pre-formatted context
+            memory_response = await self._client.thread.get_user_context(thread_id=thread_id, mode=mode)
+            context_string = memory_response.context or ""
+
+            # Return as a single MemoryItem with the formatted context
+            if context_string:
+                return [
+                    MemoryItem(conversation=[],
+                               user_id=user_id,
+                               memory=context_string,
+                               metadata={
+                                   "mode": mode, "thread_id": thread_id
+                               })
+                ]
+            else:
+                return []
+
+        except Exception as e:
+            # If thread doesn't exist or no context available, return empty list
+            if "404" in str(e) or "not found" in str(e).lower():
+                return []
+            raise
 
     async def remove_items(self, **kwargs):
+        """
+        Remove items for a specific thread.
 
-        if "session_id" in kwargs:
-            session_id = kwargs.pop("session_id")
-            await self._client.memory.delete(session_id)
-
+        Args:
+            kwargs: Additional parameters.
+                - thread_id (str): Thread ID to delete specific thread.
```

Comment:
nit: from a public interface perspective, thread id has been encapsulated internally until this point. Would it make more sense for `conversation_id` to be used as the named parameter rather than `thread_id` ?